src/test/java/org/motechproject/prevac/service/CapacityInfoServiceTest.java

```diff
@@ -0,0 +1,158 @@
+import static org.mockito.MockitoAnnotations.initMocks;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(CapacityInfoService.class)
+public class CapacityInfoServiceTest {
+
+    @InjectMocks
+    private CapacityInfoService capacityInfoService = new CapacityInfoServiceImpl();
+
+    @Mock
+    private ScreeningDataService screeningDataService;
+
+    @Mock
+    private UnscheduledVisitDataService unscheduledVisitDataService;
+
+    @Mock
+    private VisitDataService visitDataService;
+
+    @Mock
+    private ClinicDataService clinicDataService;
+
+    @Before
+    public void setUp() {
+        initMocks(this);
+    }
+
+    @Test
+    public void shouldCalculateClinicCapacity() {
+        GridSettings gridSettings = createGridSettings(1, 10, DateFilter.DATE_RANGE, "2017-1-1", "2017-1-2");
+
+        Clinic firstClinic = createClinic(1L, "siteId1", "first", 20, 5, 4, 3, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0);
+        Clinic secondClinic = createClinic(2L, "siteId2", "second", 20, 10, 2, 9, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0);
+        List<Clinic> clinics = new ArrayList<>(Arrays.asList(firstClinic, secondClinic));
+
+        Range<LocalDate> dateRange = new Range<>(new LocalDate(2017, 1, 1), new LocalDate(2017, 1, 2));
+
+        when(clinicDataService.retrieveAll(Mockito.any(QueryParams.class))).thenReturn(clinics);
+        when(visitDataService.countFindByClinicIdAndPlannedVisitDateRange(firstClinic.getId(), dateRange)).thenReturn(8L);
+        when(visitDataService.countFindByClinicIdVisitTypeAndPlannedVisitDateRange(firstClinic.getId(), VisitType.PRIME_VACCINATION_DAY, dateRange)).thenReturn(7L);
+        when(screeningDataService.countFindByClinicIdAndDateRangeAndStatus(firstClinic.getId(), dateRange, ScreeningStatus.ACTIVE)).thenReturn(4L);
+        when(unscheduledVisitDataService.countFindByClinicIdAndDateRange(firstClinic.getId(), dateRange)).thenReturn(3L);
+
+        when(visitDataService.countFindByClinicIdAndPlannedVisitDateRange(secondClinic.getId(), dateRange)).thenReturn(4L);
+        when(visitDataService.countFindByClinicIdVisitTypeAndPlannedVisitDateRange(secondClinic.getId(), VisitType.PRIME_VACCINATION_DAY, dateRange)).thenReturn(2L);
+        when(screeningDataService.countFindByClinicIdAndDateRangeAndStatus(secondClinic.getId(), dateRange, ScreeningStatus.ACTIVE)).thenReturn(6L);
+        when(unscheduledVisitDataService.countFindByClinicIdAndDateRange(secondClinic.getId(), dateRange)).thenReturn(1L);
+
+        when(clinicDataService.count()).thenReturn(2L);
+
+        List<CapacityInfoDto> capacityInfoDtos = new ArrayList<>(Arrays.asList(
+                new CapacityInfoDto("first", 10, -5, 4, -1),
+                new CapacityInfoDto("second", 20, 9, -2, 16)));
+
+        Records<CapacityInfoDto> result = capacityInfoService.getCapacityInfoRecords(gridSettings);
+        for (int i = 0; i < capacityInfoDtos.size(); i++) {
+            checkIfCapacityDtoAreSame(capacityInfoDtos.get(i), result.getRows().get(i));
```

Comment:
you can change the checkIfCapacityDtoAreSame to return true if the are equal and use assertTrue here